base/applications/cmdutils/where/where.c

```diff
@@ -0,0 +1,482 @@
+/*
+ * PROJECT:     ReactOS WHERE command
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Search executable files
+ * COPYRIGHT:   Copyright 2021 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+#include <windows.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strsafe.h>
+#include <conutils.h>
+#include "strlist.h" // strlist_...
+#include "resource.h"
+
+#define FLAG_HELP (1 << 0) // "/?"
+#define FLAG_R (1 << 1) // recursive directory
+#define FLAG_Q (1 << 2) // quiet mode
+#define FLAG_F (1 << 3) // double quote
+#define FLAG_T (1 << 4) // detailed info
+
+static DWORD s_dwFlags = 0;
+static LPWSTR s_pszRecursiveDir = NULL;
+static strlist_t s_patterns = strlist_default;
+static strlist_t s_results = strlist_default;
+static strlist_t s_pathext = strlist_default;
+
+// is it either "." or ".."?
+#define IS_DOTS(pch) \
+    (*(pch) == L'.' && ((pch)[1] == 0 || ((pch)[1] == L'.' && (pch)[2] == 0)))
+
+#define DEFAULT_PATHEXT L".com;.exe;.bat;.cmd"
+
+typedef enum WRET // return code of WHERE command
+{
+    WRET_SUCCESS = 0,
+    WRET_NOT_FOUND = 1,
+    WRET_ERROR = 2
+} WRET;
+
+static VOID WhereError(UINT nID)
+{
+    if (!(s_dwFlags & FLAG_Q)) // not quiet mode?
+        ConResPuts(StdErr, nID);
+}
+
+typedef BOOL (CALLBACK *WHERE_CALLBACK)(LPCWSTR pattern, LPCWSTR path, WIN32_FIND_DATAW *data);
+
+static BOOL
+WhereSearchGeneric(LPCWSTR pattern, LPWSTR path, BOOL bDir, WHERE_CALLBACK callback)
+{
+    LPWSTR pch;
+    size_t cch;
+    BOOL ret;
+    WIN32_FIND_DATAW data;
+    HANDLE hFind = FindFirstFileExW(path, FindExInfoStandard, &data, FindExSearchNameMatch,
+                                    NULL, 0);
+    if (hFind == INVALID_HANDLE_VALUE)
+        return TRUE; // not found
+
+    pch = wcsrchr(path, L'\\') + 1;
+    cch = MAX_PATH - (pch - path);
```

Comment:
max path?

Reply:
`MAX_PATH` is for the character count of string buffer `path`.
`cch` is remainder of string buffer.

Reply:
> MAX_PATH is for the character count of string buffer path.

Then I strongly suggest you add an extra argument to the function, that allows passing the actual size of the path buffer along.

Reply:
Added `path_len`.